### PR TITLE
Estimated frequency as string

### PR DESCRIPTION
**Pull request by Roddhes88**

### CODE REVIEW

File: src/Humanizer/DateHumanizeExtensions.cs [hunk 1]
```diff
@@ -26,5 +26,86 @@ public static string Humanize(this DateTime input, bool utcDate = true, DateTime
...
             return Configurator.DateTimeHumanizeStrategy.Humanize(input, comparisonBase, culture);
         }
+
+        /// <summary>
+        /// Extracts the month out of a date
+        /// </summary>
+        /// <param name="input">The date of which you want the month</param>
+        /// <returns></returns>
+        public static string ToMonth(this DateTime input)
+        {
```
Comment (hazzik) [inline src/Humanizer/DateHumanizeExtensions.cs]: Shouldn't it be [DateTimeFormatInfo.MonthNames](https://msdn.microsoft.com/en-us/library/system.globalization.datetimeformatinfo.monthnames) [DateTimeFormatInfo.MonthGenitiveNames](https://msdn.microsoft.com/en-us/library/system.globalization.datetimeformatinfo.monthgenitivenames)?

File: src/Humanizer/DateHumanizeExtensions.cs [hunk 2]
```diff
@@ -26,5 +26,86 @@ public static string Humanize(this DateTime input, bool utcDate = true, DateTime
...
+            }
+            return output;
+        }
+
+        /// <summary>
+        /// Extracts the day out of a date
+        /// </summary>
+        /// <param name="input">The date of which you want the day</param>
+        /// <returns></returns>
+        public static string ToDay(this DateTime input)
```
Comment (hazzik) [inline src/Humanizer/DateHumanizeExtensions.cs]: Shouldn't we use Humanizer's `.Ordinalize()` to get these numbers?

File: src/Humanizer/FrequencyEnum.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Humanizer
+{
+    public enum FrequencyEnum
```
Comment (hazzik) [inline src/Humanizer/FrequencyEnum.cs]: We don't use "Enum" postfix here

File: src/Humanizer/FrequencyHumanizeExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+{
+    public static class FrequencyHumanizeExtensions
+    {
+        public static string Humanize(this Frequency input)
+        {
+            string output = "every";
+            switch (input.OccurrenceFrequency)
+            {
+                case FrequencyEnum.Never:
+                    output = "never";
```
Comment (hazzik) [inline src/Humanizer/FrequencyHumanizeExtensions.cs]: These strings need to be translatable

File: src/Humanizer/TimeUnitEnum.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Humanizer
+{
+    public enum TimeUnitEnum
```
Comment (hazzik) [inline src/Humanizer/TimeUnitEnum.cs]: Same, no "Enum" please

File: src/Humanizer/FrequencyHumanizeExtensions.cs (hunk 4 above)
Comment (Roddhes88) [inline src/Humanizer/FrequencyHumanizeExtensions.cs]: I don't really understand what you mean by translatable? Can you explain
please?
Le 29 mars 2015 23:33, "Alexander Zaytsev" <notifications@github.com> a
écrit :

Comment (hazzik) [inline src/Humanizer/FrequencyHumanizeExtensions.cs]: I mean, to put these strings to .NET resources, so we can get translations out of the box

Comment (Roddhes88) [inline src/Humanizer/FrequencyHumanizeExtensions.cs]: Ok thank you, I've amended the other things as you told me to do, now I'll
try to find out how to do this last one, then I'll do another pull request.
 Le 30 mars 2015 00:04, "Alexander Zaytsev" <notifications@github.com> a
écrit :

Comment (hazzik) [inline src/Humanizer/FrequencyHumanizeExtensions.cs]: You don't need to do another pull request, you can push more commits here.

File: src/Humanizer/TimeUnitEnum.cs (hunk 5 above)
Comment (mexx) [inline src/Humanizer/TimeUnitEnum.cs]: There is already a `TimeUnit` enum present in `Humanizer.Localisation` namespace, see [TimeUnit.cs](https://github.com/MehdiK/Humanizer/blob/master/src/Humanizer/Localisation/TimeUnit.cs)

File: src/Humanizer.sln [hunk 6]
```diff
@@ -1,6 +1,8 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2012
```
Comment (MehdiK) [inline src/Humanizer.sln]: Can you please revert solution file changes?

File: src/Humanizer/DateHumanizeExtensions.cs (hunk 1 above)
Comment (MehdiK) [inline src/Humanizer/DateHumanizeExtensions.cs]: :+1:

File: src/Humanizer/DateHumanizeExtensions.cs (hunk 2 above)
Comment (MehdiK) [inline src/Humanizer/DateHumanizeExtensions.cs]: :+1:

File: src/Humanizer/Frequency.cs [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+        private Frequencies _occurrenceFrequency;
+        private List<DateTime> _listOfOccurrences;
+        private TimeSpan _timeSpanBetweenOccurrences;
+        private NotStandardFrequencyInfo _notStandardFrequencyInfo;
+
+        public DateTime StartDate
+        {
+            get { return _startDate; }
+            set
+            {
```
Comment (MehdiK) [inline src/Humanizer/Frequency.cs]: All setters should be private.

File: src/Humanizer/Frequency.cs [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Humanizer.Localisation;
+
+namespace Humanizer
+{
+    public class Frequency
+    {
+        private DateTime _startDate = DateTime.MinValue;
```
Comment (MehdiK) [inline src/Humanizer/Frequency.cs]: We should change all `DateTime` instances to `DateTimeOffset`.

File: src/Humanizer/Frequency.cs [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+	        {
+                NotStandardFrequencyInfo.Frequency = new KeyValuePair<TimeUnit, double>(TimeUnit.Minute, SplitTimespan(input, 60, (int)input.TotalSeconds));
+            }
+            else if ((int)input.TotalSeconds > 0)
+	        {
+                NotStandardFrequencyInfo.Frequency = new KeyValuePair<TimeUnit, double>(TimeUnit.Second, SplitTimespan(input, 1000, (int)input.TotalMilliseconds));
+            }
+        }
+
+        private double SplitTimespan(TimeSpan input, int nbUnit, int referenceAmount)
```
Comment (MehdiK) [inline src/Humanizer/Frequency.cs]: Is `nbUnit` number of units? Can we rename it to `units`?

File: src/Humanizer.Tests/FrequencyHumanizeTest.cs [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+namespace Humanizer.Tests
+{
+    public class FrequencyHumanizeTest
+    {
+
+        [Fact]
+        public void EveryTwelveHours(TimeSpan timespan, string expectedSentence)
+        {
+            Assert.True(Frequency.Every(new TimeSpan(12, 0, 0)).Humanize() == "roughly every twelve hours");
```
Comment (MehdiK) [inline src/Humanizer.Tests/FrequencyHumanizeTest.cs]: Replace `Assert.True` with `Assert.Equal`.

File: src/Humanizer.Tests/FrequencyHumanizeTest.cs [hunk 11]
```diff
@@ -0,0 +1,54 @@
+﻿using Xunit;
+using Xunit.Extensions;
+using System;
+
+namespace Humanizer.Tests
+{
+    public class FrequencyHumanizeTest
```
Comment (MehdiK) [inline src/Humanizer.Tests/FrequencyHumanizeTest.cs]: There is no direct test coverage for `Frequency`. Please unit test `Frequency`.

File: src/Humanizer.Tests/FrequencyHumanizeTest.cs [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    {
+
+        [Fact]
+        public void EveryTwelveHours(TimeSpan timespan, string expectedSentence)
+        {
+            Assert.True(Frequency.Every(new TimeSpan(12, 0, 0)).Humanize() == "roughly every twelve hours");
+        }
+
+        [Fact]
+        public void EveryThreeDaysAndAHalf(TimeSpan timespan, string expectedSentence)
```
Comment (MehdiK) [inline src/Humanizer.Tests/FrequencyHumanizeTest.cs]: I am no expert in English but I think this should read as 'roughly every three and a half days'

File: src/Humanizer.Tests/FrequencyHumanizeTest.cs [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+        public void EveryYear(TimeSpan timespan, string expectedSentence)
+        {
+            Assert.True(Frequency.Every().Year().Humanize() == "every year");
+        }
+
+        [Fact]
+        public void Every(TimeSpan timespan, string expectedSentence)
+        {
+            Assert.True(Frequency.Every().Year().StartingOn(new DateTime(2014, 2, 3)).EndingOn(new DateTime(2018, 8, 3)).Humanize() == 
+                "every 3rd of february from 3rd of february 2014 till 3rd of august 2018");
```
Comment (MehdiK) [inline src/Humanizer.Tests/FrequencyHumanizeTest.cs]: Should this read as 'every 3rd of February from 2014 till 2018'. Also I think month names should start with capital letters.

File: src/Humanizer.Tests/FrequencyHumanizeTest.cs (hunk 10 above)
Comment (MehdiK) [inline src/Humanizer.Tests/FrequencyHumanizeTest.cs]: Cool. This reads nicely using the fluent API: `Frequency.Every(2.Days())`

File: src/Humanizer.Tests/FrequencyHumanizeTest.cs [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        [Fact]
+        public void EveryTimespan(TimeSpan timespan, string expectedSentence)
+        {
+            Assert.True(Frequency.Every(new DateTime(2015, 12, 05)).Humanize() == "every year on 5th of december");
```
Comment (MehdiK) [inline src/Humanizer.Tests/FrequencyHumanizeTest.cs]: I find this usage rather confusing. I wouldn't expect the frequency of this to be yearly!

File: src/Humanizer/DateHumanizeExtensions.cs [hunk 15]
```diff
@@ -26,5 +26,86 @@ public static string Humanize(this DateTime input, bool utcDate = true, DateTime
 
             return Configurator.DateTimeHumanizeStrategy.Humanize(input, comparisonBase, culture);
         }
+
+        /// <summary>
+        /// Extracts the month out of a date
+        /// </summary>
+        /// <param name="input">The date of which you want the month</param>
+        /// <returns></returns>
+        public static string ToMonth(this DateTime input)
```
Comment (MehdiK) [inline src/Humanizer/DateHumanizeExtensions.cs]: I don't see any unit tests for this!

Comment (MehdiK) [inline src/Humanizer/DateHumanizeExtensions.cs]: Do we even need these new extensions on `DateTime`?

File: src/Humanizer/EnumHumanizeExtensions.cs [hunk 16]
```diff
@@ -74,5 +74,22 @@ public static string Humanize(this Enum input, LetterCasing casing)
 
             return humanizedEnum.ApplyCase(casing);
         }
+
+        public static string GetStringValue(this Enum value)
```
Comment (MehdiK) [inline src/Humanizer/EnumHumanizeExtensions.cs]: Why not use the existing `Description` attribute?

File: src/Humanizer/Frequency.cs [hunk 17]
```diff
@@ -0,0 +1,239 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Humanizer.Localisation;
+
+namespace Humanizer
+{
+    public class Frequency
```
Comment (MehdiK) [inline src/Humanizer/Frequency.cs]: This class exposes a big public interface but doesn't have any unit test coverage. Every public method should be extensively unit tested.

File: src/Humanizer/StringValue.cs [hunk 18]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Humanizer
+{
+    public class StringValue : System.Attribute
```
Comment (MehdiK) [inline src/Humanizer/StringValue.cs]: I'm not sold on this class! .Net has a variety of this attribute built into it and Humanizer supports them all. Why add a new one?!

**Comment by MehdiK** [comment]
I like this PR. Thanks for the contribution. 

There are a few outstanding issues on it that need to be fixed before this is merged though.

**Comment by MehdiK** [comment]
This PR has been open for two months now without any updates on it and as much as I like it I have to close it. It would be great if @Roddhes88 or anyone else could pick this up, fix the outstanding issues, rebase on the latest and submit a new PR. Closing it for now.

File: src/Humanizer.sln [hunk 19]
```diff
@@ -24,6 +26,8 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Build", "Build", "{97AAE24D
 		Humanizer.ruleset = Humanizer.ruleset
 	EndProjectSection
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Rod", "Rod\Rod.csproj", "{AB77AB8D-5EFE-43B5-BA4E-43BDF4D4E7BB}"
+EndProject
```
Comment (hazzik) [inline src/Humanizer.sln]: What is that?

**Comment by hazzik** [comment]
@MehdiK @Roddhes88 I would like to continue the discussion:)

File: src/Humanizer/Frequency.cs [hunk 20]
```diff
@@ -0,0 +1,239 @@
...
+    {
+        private DateTime _startDate = DateTime.MinValue;
+        private DateTime _endDate = DateTime.MinValue;
+        private Frequencies _occurrenceFrequency;
+        private List<DateTime> _listOfOccurrences;
+        private TimeSpan _timeSpanBetweenOccurrences;
+        private NotStandardFrequencyInfo _notStandardFrequencyInfo;
+
+        public DateTime StartDate
```
Comment (hazzik) [inline src/Humanizer/Frequency.cs]: I feel that this should be `Nullable<DateTime>`

File: src/Humanizer/FrequencyHumanizeExtensions.cs [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+using System.Globalization;
+using Humanizer.Localisation;
+using Humanizer.Configuration;
+using Humanizer.Localisation.Formatters;
+
+namespace Humanizer
+{
+    public static class FrequencyHumanizeExtensions
+    {
+        public static string Humanize(this Frequency input, CultureInfo culture = null)
```
Comment (hazzik) [inline src/Humanizer/FrequencyHumanizeExtensions.cs]: This probably be a `ToString` method of Frequency

**Comment by MehdiK** [comment]
@hazzik did you want to reopen it?

**Comment by hazzik** [comment]
@MehdiK not at the current staget. There are far lot more issues with this PR. Even tests added do not work, and produce incorrect results.

File: src/Humanizer.Tests/FrequencyHumanizeTest.cs [hunk 22]
```diff
@@ -0,0 +1,54 @@
...
+using Xunit.Extensions;
+using System;
+
+namespace Humanizer.Tests
+{
+    public class FrequencyHumanizeTest
+    {
+
+        [Fact]
+        public void EveryTwelveHours(TimeSpan timespan, string expectedSentence)
```
Comment (hazzik) [inline src/Humanizer.Tests/FrequencyHumanizeTest.cs]: [Fact] methods should not have any parameters
